addons/casings/functions/fnc_createCasing.sqf

```diff
@@ -60,18 +60,22 @@ private _pos = _unitPos
 
 [
     {
-        params ["_modelPath", "_pos"];
+        params ["_modelPath", "_pos", "_unitPos"];
 
-        private _lisPos = (lineIntersectsSurfaces [_pos, _pos vectorAdd [0,0,-1e11], objNull, objNull, true, 1, "ROADWAY", "FIRE"]) #0;
-        private _casing = createSimpleObject [_modelPath, (_lisPos #0 vectorAdd [0,0,0.005]), true];
+        private _lisPos = (lineIntersectsSurfaces [_pos, _pos vectorAdd [0,0,-1e11], objNull, objNull, true, 1, "ROADWAY", "FIRE"]) select 0;
+        private _casingPos = (_lisPos select 0) vectorAdd [0,0,0.005];
+        if (((ASLtoATL _casingPos) select 2) < 0) then { // prevent casings sinking into the terrain
+            _casingPos set [2, _unitPos select 2];
```

Comment:
What about using `getTerrainHeightASL` instead of the unit's position's z component? I'm not sure if it's compatible with changes done by [`setTerrainHeight`](https://community.bistudio.com/wiki/setTerrainHeight), but I feel it would be more accurate, as I have doubts your solution works well in uneven terrain.
Btw, does this currently account for being in buildings? I imagine so, but I just want to be sure.

Reply:
> Buildings?

Yes. Only change is preventing a negative Z component in relation to the terrain.

> getTerrainHeightASL

Didn't know that existed, good to know.

Reply:
I tested with `_casingPos set [2, (getTerrainHeightASL _casingPos) vectorAdd [0,0,0.005]];`, but it didn't yield good results.